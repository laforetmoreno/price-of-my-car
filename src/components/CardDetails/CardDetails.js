import React from 'react';
import { objectOf, string, oneOfType, number } from 'prop-types';

import CardItem from '../CartItem';

import style from './CardDetails.module.scss';

const CardDetails = ({ details }) => (
  <ul className={style.wrapper}>
    <CardItem label="Marca" field={details?.brand} />
    <CardItem label="Modelo" field={details?.model} />
    <CardItem label="Versão" field={details?.version} />
    <CardItem label="Ano" field={details?.modelYear} />
    <CardItem label="Km médio" field={details?.kmMedio} />
    <CardItem price label="Preço tabela FIPE" field={details?.precoFipe} />
    <CardItem price label="Preço mínimo:" field={details?.precoMinimo} />
    <CardItem price label="Preço médio:" field={details?.precoMedio} />
    <CardItem price label="Preço máximo:" field={details?.precoMaximo} />
    <CardItem price label="Piso preço médio:" field={details?.pisoPrecoMedio} />
    <CardItem price label="Piso preço médio volanty:" field={details?.pisoPrecoMedioVolanty} />
    <CardItem price label="Teto preço médio volanty:" field={details?.tetoPrecoMedioVolanty} />
    <CardItem price label="Teto preço concessionária:" field={details?.tetoPrecoConcessionaria} />
    <CardItem price label="Piso preço concessionária:" field={details?.pisoPrecoConcessionaria} />
  </ul>
);

CardDetails.propTypes = {
  details: objectOf(oneOfType([string, number])),
};

export default CardDetails;
