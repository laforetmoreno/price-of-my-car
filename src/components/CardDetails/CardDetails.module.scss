.wrapper {
  display: flex;
  margin: 40px 0;
  align-items: center;
  list-style: none;
  padding: 0;
  flex-wrap: wrap;
  box-shadow: 0 0 2rem 0 rgba(136, 152, 170, 0.15);
  padding: 25px;
  border-radius: 4px;

  @media (max-width: 767px) {
    flex-direction: column;
  }
}
