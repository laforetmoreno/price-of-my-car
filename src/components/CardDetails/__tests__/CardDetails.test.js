import React from 'react';
import renderer from 'react-test-renderer';
import CardDetails from '../CardDetails';

describe('<CardDetails />', () => {
  it('renders correctly', () => {
    const fakeCar = {
      precoFipe: 50639,
      precoMinimo: 43900,
      precoMedio: 50596,
      tetoPrecoMedio: 49584,
      pisoPrecoMedio: 48066,
      tetoPrecoMedioVolanty: 46042,
      pisoPrecoMedioVolanty: 44524,
      precoMaximo: 59990,
      tetoPrecoConcessionaria: 40477,
      pisoPrecoConcessionaria: 35417,
      kmMedio: 24647,
      qtd: 174,
      brand: 'VOLKSWAGEN',
      model: 'POLO',
      version: '1.6 MSI TOTAL FLEX MANUAL',
      modelYear: 2018,
      versionId: '701f0eb7764832fdce094a592cdaf38c',
    };

    const tree = renderer.create(<CardDetails car={fakeCar} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
