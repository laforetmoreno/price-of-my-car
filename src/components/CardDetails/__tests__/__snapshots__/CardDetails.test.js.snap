// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardDetails /> renders correctly 1`] = `
<ul
  className="wrapper"
>
  <li
    className="item"
  >
    <span>
      Marca
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Modelo
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Versão
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Ano
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Km médio
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Preço tabela FIPE
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Preço mínimo:
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Preço médio:
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Preço máximo:
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Piso preço médio:
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Piso preço médio volanty:
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Teto preço médio volanty:
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Teto preço concessionária:
    </span>
    <strong>
      -
    </strong>
  </li>
  <li
    className="item"
  >
    <span>
      Piso preço concessionária:
    </span>
    <strong>
      -
    </strong>
  </li>
</ul>
`;
